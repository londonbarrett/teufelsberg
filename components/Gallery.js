import * as React from 'react';
import PropTypes from 'prop-types';
import Thumbnail from './Thumbnail';

const Gallery = ({
  title, images,
}) => {
  const imagesList = images.map(value => (
    <Thumbnail uri={value} alt={title} key={value} />
  ));
  return (
    <div>
      <h1>{title}</h1>
      <div>
        {imagesList}
      </div>
    </div>
  );
};

Gallery.propTypes = {
  title: PropTypes.string.isRequired,
  images: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Gallery;
