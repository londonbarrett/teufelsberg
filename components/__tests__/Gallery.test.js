/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Gallery from '../Gallery';

describe('Gallery Tests', () => {
  it('Renders correctly', () => {
    const images = [
      'image_01',
      'image_02',
      'image_03',
      'image_04',
    ];
    const component = renderer.create((
      <Gallery
        title="Image Gallery"
        images={images}
      />
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
