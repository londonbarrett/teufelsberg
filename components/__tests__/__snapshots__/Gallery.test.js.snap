// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Tests Renders correctly 1`] = `
<div>
  <h1>
    Image Gallery
  </h1>
  <div>
    <a
      href="/photo?id=image_01"
      onClick={[Function]}
    >
      <img
        alt="Image Gallery"
        src="https://image_01_2.jpg"
      />
    </a>
    <a
      href="/photo?id=image_02"
      onClick={[Function]}
    >
      <img
        alt="Image Gallery"
        src="https://image_02_2.jpg"
      />
    </a>
    <a
      href="/photo?id=image_03"
      onClick={[Function]}
    >
      <img
        alt="Image Gallery"
        src="https://image_03_2.jpg"
      />
    </a>
    <a
      href="/photo?id=image_04"
      onClick={[Function]}
    >
      <img
        alt="Image Gallery"
        src="https://image_04_2.jpg"
      />
    </a>
  </div>
</div>
`;
